app/helpers/users_helper.rb

```diff
@@ -24,4 +24,18 @@ def user_taxonomies_html_options(user)
       :organization => { :onchange => 'taxonomy_added(this, "organization")'}
     } unless user.admin?
   end
+
+  def mail_notification_settings(mail_notification)
+    (_(mail_notification.description) + mail_notification_query_builder_settings(mail_notification)).html_safe
+  end
+
+  def mail_notification_query_builder_settings(mail_notification)
+    mail_notification.queryable? ? link_to_function(icon_text('cog', "", { :class => "#{mail_notification.name}_query_setting",
+                                                                           :title => _('Add query to this notification')}),
```

Comment:
I wouldn't advise using "query" in the UI, we generally use "search" or "filter".